TemporaryProjects.14.0/TempProjectsLocationContext.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.IO;
+using EnvDTE;
+using Microsoft.VisualStudio.Shell;
+
+namespace TemporaryProjects
+{
+    public class TempProjectsLocationContext : IDisposable
```

Comment:
I would personally prefer a struct like SetPropertyValueContext (or if we were to be extra fancy, a factory extension method on a DTE property :smile:), which would take a DTE property and a desired value and would set it to the original value on Dispose. I thought about doing this in the abandonded PR because it could have been reused for both settings the project location and setting SaveNewProjects.

Reply:
I'm curious why a struct?

Reply:
I guess it doesn't matter because this is far from being performance critical. I'd just default to a struct though for a trivial wrapper like this.